### PR TITLE
Handle None values in dataset and resource extras endpoints

### PR DESCRIPTION
**Pull request by abulte**
Fix https://github.com/etalab/data.gouv.fr/issues/1001

- If key does not exist: ignore
- If key exists: delete

### CODE REVIEW

**Review by quaxsze — APPROVED** [review]

File: udata/tests/apiv2/test_datasets.py [hunk 1]
```diff
@@ -199,22 +199,30 @@ def test_get_dataset_extras(self):
...
         data = ['test::extra-second', 'another::key']
         response = self.put(url_for('apiv2.dataset_extras', dataset=self.dataset), data)
         self.assert400(response)
         assert response.json['message'] == 'Wrong payload format, dict expected'
 
-        data = {'test::extra-second': 'test-value-changed', 'another::key': 'another-value'}
+        data = {
+            'test::extra-second': 'test-value-changed',
+            'another::key': 'another-value',
+            'test::none': None,
```
Comment (maudetes) [inline udata/tests/apiv2/test_datasets.py:216]: I don't think we test that it unsets a key here, right?

**Review by maudetes — COMMENTED** [review]
Thank you!
I've made some minor comments :)

File: udata/tests/apiv2/test_datasets.py (hunk 1 above)
Comment (abulte) [reply udata/tests/apiv2/test_datasets.py:217]: True! I added a test.

**Review by maudetes — APPROVED** [review]
Nice! :clap:

**Pull request merged**